Move Queue related stuff into EngagementRepository

DavDo opened the pull request:
**Jira issue:**
https://glia.atlassian.net/browse/MOB-2950

**What was solved?**

I moved Queue-related stuff to the EngagementRepository. This is the final step collecting all the engagement-related stuff in one place. So now all the events related to the Engagement are handled in EngagementRepository. I also fixed `Queueing` state for the chat bubble.

**Release notes:**

 - [ ] Feature
 - [x] Ignore
 - [ ] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

**Additional info:**

 - [ ] Tests updated? Added? Unit, acceptance, snapshots?
 - [ ] Logging for future troubleshooting of client issues added?

**Screenshots:**

andrews-moc commented on widgetssdk/src/main/java/com/glia/widgets/call/CallController.kt:
```diff
@@ -229,12 +221,8 @@ internal class CallController(
-                .execute(queueId, visitorContextAssetId, mediaType)
-                .subscribe({ queueForEngagementStarted() }) { queueForEngagementError(it) }
-        )
+    private fun queueForEngagement(queueId: String, visitorContextAssetId: String?, mediaType: Engagement.MediaType) {
+        queueForEngagementUseCase(queueId, mediaType, visitorContextAssetId)
```
Not critical. I would suggest to rename using some verb. For example, `startQueueForEngagement()`, `startQueueForEngagementUseCase()` because otherwise it's a bit unclear if it checks for queueing state or initiates it.

andrews-moc commented on widgetssdk/src/main/java/com/glia/widgets/engagement/EngagementRepositoryImpl.kt:
```diff
@@ -148,9 +207,9 @@ internal class EngagementRepositoryImpl(
     override fun declineCurrentEngagementRequest() {
         _engagementRequest.firstOrError().flatMapCompletable(::decline).subscribe({
-            Logger.i(TAG, "Incoming Call Visualizer engagement was declined")
+            i(TAG, "Incoming Call Visualizer engagement was declined")
```
I would suggest to keep the full reference: `Logger.d()`, `Logger.i()`, `Logger.e()`, like we have in the majority of places. Keeping single style makes search easier.

andrews-moc commented on widgetssdk/src/main/java/com/glia/widgets/helper/CommonExtensions.kt:
```diff
@@ -69,7 +71,13 @@ internal val GliaException.isQueueUnavailable: Boolean
+
 @SuppressLint("CheckResult")
 internal fun <T> Single<out T>.unSafeSubscribe(onNextCallback: (T) -> Unit) {
     subscribe({ onNextCallback(it) }) { Logger.e("Single<T>.unSafeSubscribe", "Unexpected Local exception happened", it) }
 }
+
+@SuppressLint("CheckResult")
+internal fun Completable.unSafeSubscribe(onComplete: Action) {
+    subscribe(onComplete) { Logger.e("Single<T>.unSafeSubscribe", "Unexpected Local exception happened", it) }
```
NIT: Local => local

andrews-moc commented on widgetssdk/src/main/java/com/glia/widgets/view/head/controller/ServiceChatHeadController.kt:
```diff
@@ -107,10 +104,22 @@ internal class ServiceChatHeadController(
-        } else if (state is FinishedCallVisualizer || state is FinishedOmniCore) {
-            engagementEnded()
+        when (state) {
+            is EngagementState.StartedOmniCore, is EngagementState.StartedCallVisualizer -> {
+                newEngagementLoaded()
+            }
+
+            is EngagementState.FinishedCallVisualizer, is EngagementState.FinishedOmniCore, is EngagementState.QueueUnstaffed, is EngagementState.UnexpectedErrorHappened, is EngagementState.QueueingCanceled -> {
+                engagementEnded()
+            }
+
+            is EngagementState.Queuing, is EngagementState.PreQueuing -> {
+                queueingStarted()
+            }
```
It might be easier for reading if each state starts from the next line, like in the `ApplicationChatHeadLayoutController`.

andrews-moc reviewed (approved):
Great job, really. 👍

DavDo commented:
@andrews-moc Thank you for reviewing, I addressed the comments and also moved the `Unexpected error` and `Unstaffed queue` dialogs to `EngagementCompletionController`

andrews-moc commented:
@DavDo New changes are looking good to me. 👍
